### PR TITLE
SQL-1197: Change types for all

### PR DESCRIPTION
**Pull request by terakilobyte**
This returns the table with the changed types.

It also includes a lot of automatic formatting. I've commented the non-formatting changes.

### CODE REVIEW

File: connector/MongoDBAtlasODBC.pq [hunk 1]
```diff
@@ -1,60 +1,62 @@
...
+                            "Data",
+                            (tables) => TransformColumnKeepType(tables, "Data", TransformSqlUnknownColumns)
+                        )
             in
-                TransformColumns,
-        WithMetadata = Value.ReplaceType(TransformColumns, Value.ReplaceMetadata(Value.Type(DataSource), Metadata))
+                TransformColumns
     in
-        WithMetadata;
+        TransformColumns;
```
Comment (terakilobyte) [inline connector/MongoDBAtlasODBC.pq:59]: Remove `WithMetadata` entirely, and the Value.ReplaceType operation here.

File: connector/MongoDBAtlasODBC.pq [hunk 2]
```diff
@@ -67,36 +69,42 @@ GetCredentialConnectionString = () as record =>
...
+        SqlBinaryTypes = Table.ToRecords(
+            Table.SelectRows(Schema, each [TypeName] = "Binary.Type" and [NativeTypeName] <> "binData")
+        ),
+        TransformOperations = List.Accumulate(
+            SqlBinaryTypes,
+            table,
+            (state, column) =>
+                let
+                    TransformColumn = Table.TransformColumns(
+                        state, {column[Name], Text.FromBinary, type text}, null, MissingField.UseNull
```
Comment (terakilobyte) [inline connector/MongoDBAtlasODBC.pq:84]: Specifying a default type for the changed columns

File: connector/MongoDBAtlasODBC.pq [hunk 3]
```diff
@@ -67,36 +69,42 @@ GetCredentialConnectionString = () as record =>
...
 // Executes operation on specified columnName of the passed in table
 shared TransformColumnKeepType = (table, columnName, operation) =>
     let
-        TableType = Value.Type(table),
-        TransformColumn = Table.TransformColumns(table, { columnName, operation}),
-        ReplaceType = Value.ReplaceType(TransformColumn, TableType)
+        TableType = Value.Type(table), TransformColumn = Table.TransformColumns(table, {columnName, operation})
     in
-        ReplaceType;
+        Value.ReplaceType(TransformColumn, TableType);
```
Comment (terakilobyte) [inline connector/MongoDBAtlasODBC.pq:97]: Removed the assign and returned the operation directly.

File: connector/MongoDBAtlasODBC.pq (hunk 2 above)
Comment (nbagnard) [reply connector/MongoDBAtlasODBC.pq:84]: It sets the column type as PBI Text to make sure that PBI will show the right icon and set of transform according to the type.
This is basically the only change, right?
Everything else is refactoring in one way or another

Comment (terakilobyte) [reply connector/MongoDBAtlasODBC.pq:84]: Correct!

**Review by nbagnard — APPROVED** [review]
LGTM! :)

File: connector/MongoDBAtlasODBC.pq [hunk 4]
```diff
@@ -80,9 +80,13 @@ shared TransformSqlUnknownColumns = (table) =>
             table,
             (state, column) =>
                 let
-                    TransformColumn = Table.TransformColumns(
-                        state, {column[Name], Text.FromBinary, type text}, null, MissingField.UseNull
-                    )
+                    ColumnType = Type.TableColumn(Value.Type(table), column[Name]),
+                    PreservedFacetFields = {"NativeTypeName", "NativeDefaultExpression", "NativeExpression"},
+                    Facets = Record.SelectFields(Type.Facets(ColumnType), PreservedFacetFields),
+                    ComplexType = column[NativeTypeName],
```
Comment (bucaojit) [inline connector/MongoDBAtlasODBC.pq:86]: This variable `ComplexType` isn't used, it can be removed.

**Review by bucaojit — APPROVED** [review]
LGTM!  It's great you were able to preserve the native type name.

**Comment by terakilobyte** [comment]
evergreen merge

**Pull request merged**